src/pages/TermsOfService.test.ts

```diff
@@ -115,4 +148,47 @@ describe('TermsOfService', () => {
     const acceptButton = screen.queryByText('Accept');
     expect(acceptButton).not.toBeInTheDocument();
   });
+  it('calls the acceptTos endpoint when the accept tos button is clicked', async () => {
+    // Arrange
+    const termsOfService = {
+      userHasAcceptedLatestTos: false,
+      permitsSystemUsage: true,
+    };
+
+    const { acceptTosFn } = setupMockAjax(termsOfService);
+
+    // Act
+    await act(async () => { render(h(TermsOfServicePage)) }) //eslint-disable-line
+
+    // Assert
+    const acceptButton = screen.queryByText('Accept');
+    expect(acceptButton).toBeInTheDocument();
+    expect(acceptButton).not.toBeNull();
+    if (acceptButton) {
+      await act(async () => acceptButton.click());
+    }
+    expect(acceptTosFn).toHaveBeenCalled();
```

Comment:
If the button is expected to be present, let's test for it with `getBy...`. That way, an exception is thrown if it's not found and we don't have to deal with TS thinking `acceptButton` might be null.
https://testing-library.com/docs/queries/about#types-of-queries

`queryBy...` is basically _only_ useful for checking if things _don't_ exist.
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence

Also, suggest using `fireEvent` or `userEvent` to click.

https://testing-library.com/docs/dom-testing-library/api-events
```suggestion
    const acceptButton = screen.getByText('Accept');
    await act(async () => fireEvent.click(acceptButton));
    expect(acceptTosFn).toHaveBeenCalled();
```

or

https://testing-library.com/docs/user-event/intro
```suggestion
    const user = userEvent.setup();
    ...
    const acceptButton = screen.getByText('Accept');
    await user.click(acceptButton);
    expect(acceptTosFn).toHaveBeenCalled();
```